spec/system/common_area/reservation/resident_reserves_common_area_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+describe 'Resident reserves common area' do
+  it 'only if authenticated' do
```

Comment:
Poderia haver um teste de um morador que não possui acesso a esse condomínio tentando forçar essa rota e ser redirecionado.

Reply:
Criamos um novo teste e implementamos as configurações para restringir que um 'resident' que não tenha residência no condomínio da área comum não consiga acessar a página de reserva e seja redirecionado para o root_path.